client/app/services/query.js

```diff
@@ -35,13 +35,14 @@ class Parameters {
     this.initFromQueryString(queryString);
   }
 
-  parseQuery() {
+  parseQuery(queryText = null) {
     const fallback = () => map(this.query.options.parameters, i => i.name);
+    queryText = queryText || this.query.query;
 
     let parameters = [];
-    if (this.query.query !== undefined) {
+    if (queryText !== undefined) {
```

Comment:
```suggestion
    if (!isNil(queryText)) {
```

(need to import `isNil` from lodash)